m4/curl-rustls.m4

```diff
@@ -152,6 +152,16 @@ if test "x$OPT_RUSTLS" != xno; then
     dnl Rustls lib to LIBS
     LIBS="-lrustls -lpthread -ldl -lm $LIBS"
   fi
+  AC_CHECK_LIB(rustls, rustls_supported_hpke,
+    [
+    AC_DEFINE(USE_RUSTLS, 1, [if Rustls is enabled])
+    RUSTLS_ENABLED=1
+    USE_RUSTLS="yes"
+    ssl_msg="rustls"
+    test rustls != "$DEFAULT_SSL_BACKEND" || VALID_DEFAULT_SSL_BACKEND=yes
+    ],
+    AC_MSG_ERROR([--with-rustls was specified but could not find Rustls.]),
+    $RUSTLS_LDFLAGS)
```

Comment:
I lifted this from similar logic on L91.

When using `--prefix` with a `cargo cinstall`'d directory the 0.14 release has a pkg-config dir and so `PKGTEST="yes"` is set and the check on L91 doesn't get run.

There might be a nicer way to tidy this all up but I'm pretty adverse to mucking with m4 personally.